[quantization] Adding docstrings for nnq.functional

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27363 [quantization] Adding docstrings for nnq.functional**

Differential Revision: [D17758907](https://our.internmc.facebook.com/intern/diff/D17758907)

Commit:
[quantization] Adding docstrings for nnq.functional

raghuramank100 commented on torch/nn/quantized/functional.py:
```diff
@@ -120,6 +118,78 @@ def conv2d(input, weight, bias,
+            Default: ``False``
+    """
+    if not input.is_quantized:
+        raise ValueError("Input to 'quantized.adaptive_avg_pool2d' must be quantized!")
```
NIT: should refer to interpolate

raghuramank100 commented on torch/nn/quantized/functional.py:
```diff
@@ -120,6 +118,78 @@ def conv2d(input, weight, bias,
+
+    .. note::
+
+      Current implementation uses packed weights. This has penalty on performance.
```
Better to say, current implementation packs weights for every call.

raghuramank100 commented on torch/nn/quantized/functional.py:
```diff
@@ -120,6 +118,78 @@ def conv2d(input, weight, bias,
+        zero_point = input.q_zero_point()
+    _packed_params = torch.ops.quantized.linear_prepack(weight, bias)
+    return torch.ops.quantized.linear(input, _packed_params, scale, zero_point)
+
```
Might be good to say that maxpool2d also propagates the quantization parameters to the output

raghuramank100 replied:
Do we also want to check if the input to maxpool2d is quantized here?

Commit:
Update on "[quantization] Adding docstrings for nnq.functional"


Differential Revision: [D17758907](https://our.internmc.facebook.com/intern/diff/D17758907)

Commit:
Update on "[quantization] Adding docstrings for nnq.functional"


Differential Revision: [D17758907](https://our.internmc.facebook.com/intern/diff/D17758907)
